cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,63 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import HomePageObject from "../support/pages/home.pageObject";
+import SignUpPageObject from "../support/pages/signUp.pageObject";
+
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
+  });
 
+  it('should provide an ability to register user with valid value', () => {
+    signUpPage.visit();
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword('12345Qwert!');
+    signUpPage.clickOnSignUp();
+    homePage.checkRegistration();
+    homePage.usernameLink.should('contain', user.username);
   });
 
-  it('should ...', () => {
+  it('should not provide an ability to register user without username', () => {
+    signUpPage.visit();
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword('12345Qwert!');
```

Comment:
Hardcoded password '12345Qwert!' is used in the tests. Use generated secure passwords instead to improve the quality of tests.